### PR TITLE
[fix](java-udf)Fix need to restart BE after replacing the jar package in java-udf

### PR DESCRIPTION
**Pull request by DongLiang-0**
## Proposed changes
Previously, jar positioning was based on the pid of the file, which will result in new or modified classes in the jar, the BE cannot locate the real jar package.

Reproduction steps:
1. Create a function fun1()
2. Replace with new jar package
3. Drop fun1() function
4. Create a new fun1() and use the new jar package

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by DongLiang-0** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by DongLiang-0** [comment]
run buildall

File: be/src/runtime/user_function_cache.cpp [hunk 1]
```diff
@@ -216,10 +216,12 @@ Status UserFunctionCache::_get_cache_entry(int64_t fid, const std::string& url,
     std::string file_name = _get_file_name_from_url(url);
     {
         std::lock_guard<std::mutex> l(_cache_lock);
-        auto it = _entry_map.find(fid);
-        if (it != _entry_map.end()) {
-            entry = it->second;
-        } else {
+        for (const auto& item : _entry_map) {
+            if (item.second->checksum == checksum) {
```
Comment (zhangstar333) [inline be/src/runtime/user_function_cache.cpp:220]: seems use func id or checksum , it's also need restart.
maybe we should remove all cache jar under lib/udf folder when start BE.

Comment (DongLiang-0) [reply be/src/runtime/user_function_cache.cpp:220]: If you locate the jar based on the `checksum` parameter, there is no need to restart Be. This `checksum` is generated by the Fe end parsing the new jar package when creating the udf function.
refer to：`org.apache.doris.analysis.CreateFunctionStmt.computeObjectChecksum()`

**Review by morningman — COMMENTED** [review]
Please add test case for it

**Comment by DongLiang-0** [comment]
> Please add test case for it

Of course, I've added.

**Comment by DongLiang-0** [comment]
run buildall

File: be/src/runtime/user_function_cache.cpp [hunk 2]
```diff
@@ -216,10 +216,12 @@ Status UserFunctionCache::_get_cache_entry(int64_t fid, const std::string& url,
...
+                entry = item.second;
```
Comment (yiguolei) [inline be/src/runtime/user_function_cache.cpp:221]: add break here?

**Comment by DongLiang-0** [comment]
run buildall

File: be/src/runtime/user_function_cache.cpp (hunk 2 above)
Comment (DongLiang-0) [reply be/src/runtime/user_function_cache.cpp:221]: Thanks for your suggestion, I changed it.

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

**Pull request closed without merging**